esprima.js

```diff
@@ -674,6 +677,19 @@
 
         // Check for most common single-character punctuators.
         case 0x2E:  // . dot
+            ch3 = source.substr(index, 3);
```

Comment:
It's either substr, or charCodeAt followed by potentially substr. Is charCodeAt that much faster that it's worth doing?